ccsrc/lib/cppsim/CMakeLists.txt

```diff
@@ -0,0 +1,305 @@
+cmake_minimum_required(VERSION 3.20 FATAL_ERROR)
+
+project(
+  cppsim
+  VERSION 1.0.0
+  LANGUAGES C CXX)
+
+# Minimum required Python version (used both in this file and in the installed CMake configuration)
+set(CPPSIM_PYTHON_VERSION_MIN 3.7.0)
+
+option(BUILD_TESTING "Build the test suite?" OFF)
+
+if(NOT DEFINED ITERATOR_DEBUG_VALUE)
+  set(ITERATOR_DEBUG_VALUE 0)
+endif()
+option(MSVC_ITERATOR_DEBUG "Define _ITERATOR_DEBUG_LEVEL (defaults to 0, can be set using ITERATOR_DEBUG_VALUE)" OFF)
+
+# ==============================================================================
+
+include(GNUInstallDirs)
+
+set(CPPSIM_INSTALL_BINDIR "${CMAKE_INSTALL_BINDIR}")
+set(CPPSIM_INSTALL_SBINDIR "${CMAKE_INSTALL_SBINDIR}")
+set(CPPSIM_INSTALL_SYSCONFDIR "${CMAKE_INSTALL_SYSCONFDIR}")
+set(CPPSIM_INSTALL_INCLUDEDIR "${CMAKE_INSTALL_INCLUDEDIR}/cppsim")
+set(CPPSIM_INSTALL_DATADIR "${CMAKE_INSTALL_DATADIR}/cppsim")
+set(CPPSIM_INSTALL_LIBDIR "${CMAKE_INSTALL_LIBDIR}/cppsim")
+set(CPPSIM_INSTALL_DOCDIR "${CMAKE_INSTALL_DATADIR}/doc/cppsim")
+set(CPPSIM_INSTALL_CMAKEDIR "${CPPSIM_INSTALL_DATADIR}/cmake")
+set(CPPSIM_INSTALL_3RDPARTYDIR "${CPPSIM_INSTALL_LIBDIR}/third_party")
+
+# ==============================================================================
+
+# TODO
+include("${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake/compiler_has_std_filesystem.cmake")
+
+# ==============================================================================
+
+include(FetchContent)
+set(FETCHCONTENT_QUIET OFF)
+
+if(BUILD_TESTING)
+  FetchContent_Declare(
+    googletest
+    GIT_REPOSITORY https://github.com/google/googletest.git
+    GIT_TAG 25cc5777a17820a6339204a3552aa1dd5e428669)
+  FetchContent_Declare(
+    hipThrust
+    GIT_REPOSITORY https://github.com/dmikushin/Thrust.git
+    GIT_TAG 9a12c1259805ed0a3a5fe9bdeb098a872deb936b)
+  FetchContent_MakeAvailable(hipThrust googletest)
+
+  if(NOT DEFINED Eigen3_DIR)
+    FetchContent_Declare(
+      Eigen3
+      GIT_REPOSITORY https://gitlab.com/libeigen/eigen.git
+      GIT_TAG 3147391d946bb4b6c68edd901f2add6ac1f31f8c)
+    FetchContent_MakeAvailable(googletest Eigen3)
+  else()
+    find_package(Eigen3 CONFIG REQUIRED)
+  endif()
```

Comment:
This should be removed since Eigen3 is found by MindQuantum in its third-party libraries directory.

You can add a short comment saying that one _could want_ to look for `Eigen3` here if it was not before.